src/ctapipe_io_nectarcam/__init__.py

```diff
@@ -92,16 +92,172 @@ class PixelStatus(IntFlag):
     num_mirror_tiles=86,  # Garczarczyk 2017
 )
 
+def module_central(geometry):
+    '''
+    Input:
+        camera geometry  
+            e.g. `source.subarray.tel[10].camera.geometry`
+    Returns:
+        pix_mid_module
+            list of pixels at the centre of each module (sorted)
+        
+    Requires the geometry to have pixel_x and pixel_y, and to have 7-pixel modules.
+    Neighbours gets lazy-loaded by the main ctapipe function in camera.geometry.
+    
+    Find central pixels of all the modules, as follows:
```

Comment:
If I undestand the algorithm described here correctly, there should be a much simpler solution than the loops / argsort on dicts below by using features of the `CameraGeometry` and some numpy masking.

Let me try to see if that works

Reply:
Sure, thanks!

If it works, I can time it...

Reply:
But this is only executed once, so timing is not so important.

Reply:
```
def find_central_pixels(cam: CameraGeometry):
    # distance to the camera center
    distance = np.hypot(cam.pix_x, cam.pix_y).to_value(cam.pix_x.unit)
    
    # pixel indices sorted from farthest to closest to the center
    order = np.argsort(distance)[::-1]
    
    # While not all pixels are visited, do
    # * Next pixel center is the furthest pixel from the center that has 6 neighbors
    # * Remove that pixel and it's neighbors from the calculation
    central_pixel_ids = []    

    neighbors = cam.neighbor_matrix.copy()
    n_neighbors = neighbors.sum(axis=0)
    n_unassigned = cam.n_pixels
    
    while n_unassigned > 0:
        # Next pixel center is the furthest pixel from the center that has 6 neighbors
        center = order[(n_neighbors[order] == 6)][0]
        
        central_pixel_ids.append(center)     
        
        # remove that pixel and the pixels that belong to this module, i.e. the neighbors
        # of the central pixel, from the calculation
        other = neighbors[center]       
        n_neighbors[center] -= 7
        n_neighbors -= neighbors[other].sum(axis=0)
        
        neighbors[:, center] = False
        neighbors[center, :] = False
        neighbors[:, other] = False
        neighbors[other, :] = False
        
        n_unassigned -= 7
        
    central_pixel_ids = np.array(central_pixel_ids)
    central_pixel_ids.sort()
    
    return central_pixel_ids
```


Runs in ~22 μs on my machine

Reply:
Done.

Reply:
> Nice, about 15 times faster than mine. Not sure why...

I think you've got a unit error there... I timed both versions and got 

300 ms (milli) vs. 22 µs (micro) ;)

Reply:
Strange, I get:
```
%%timeit -n 10
pix_mid_module = module_central(geom)
```
> 231 ms ± 6.49 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
and
```
%%timeit -n 10
pix_mid_module = find_central_pixels(geom)
```
> 14.6 ms ± 2.53 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

Maybe there is some caching somewhere?